kroxylicious-api/pom.xml

```diff
@@ -72,6 +72,10 @@
             <artifactId>junit-jupiter-params</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```

Comment:
The aim is to keep the use of the netty an implementational concern known only to the internals of the Proxy.  So we don't want to bleed Netty API into the Filter API.     So my suggestion is that we have our own ClientAuth enum and use that in the API.  Let the implementation map from that to the Netty one.

Reply:
Thanks Keith, makes sense, will sort that out

Reply:
Hi @k-wall I've done an update there to have a mapper using an Enum in the api that maps between and removes that dependency. Is that closer to what you had in mind?